filament/src/fg/ResourceAllocator.h

```diff
@@ -43,23 +51,91 @@ class ResourceAllocator {
 
     void destroyRenderTarget(backend::RenderTargetHandle h) noexcept;
 
-    backend::TextureHandle createTexture(
-            backend::SamplerType target,
+    backend::TextureHandle createTexture(const char* name, backend::SamplerType target,
             uint8_t levels,
-            backend::TextureFormat format,
-            uint8_t samples,
-            uint32_t width,
-            uint32_t height,
-            uint32_t depth,
-            backend::TextureUsage usage) noexcept;
+            backend::TextureFormat format, uint8_t samples, uint32_t width, uint32_t height,
+            uint32_t depth, backend::TextureUsage usage) noexcept;
 
     void destroyTexture(backend::TextureHandle h) noexcept;
 
+    void gc() noexcept;
+
 private:
+    // TODO: these should be settings of the engine
+    static constexpr size_t CACHE_CAPACITY = 64u << 20u;   // 64 MiB
+    static constexpr size_t CACHE_MAX_AGE  = 30u;          // 64 MiB
+
+    struct TextureKey {
+        const char* name; // doesn't participate in the hash
+        backend::SamplerType target;
+        uint8_t levels;
+        backend::TextureFormat format;
+        uint8_t samples;
+        uint32_t width;
+        uint32_t height;
+        uint32_t depth;
+        backend::TextureUsage usage;
+
+        size_t getSize() const noexcept;
+
+        bool operator==(const TextureKey& other) const noexcept {
+            return target == other.target &&
+                   levels == other.levels &&
+                   format == other.format &&
+                   samples == other.samples &&
+                   width == other.width &&
+                   height == other.height &&
+                   depth == other.depth &&
+                   usage == other.usage;
+        }
+
+        friend size_t hash_value(TextureKey const& k) {
+            size_t seed = 0;
+            utils::hash::hash_combine(seed, k.target);
+            utils::hash::hash_combine(seed, k.levels);
+            utils::hash::hash_combine(seed, k.format);
+            utils::hash::hash_combine(seed, k.samples);
+            utils::hash::hash_combine(seed, k.width);
+            utils::hash::hash_combine(seed, k.height);
+            utils::hash::hash_combine(seed, k.depth);
+            utils::hash::hash_combine(seed, k.usage);
+            return seed;
+        }
+    };
+
+    struct TextureCachePayload {
+        backend::TextureHandle handle;
+        size_t age = 0;
+        uint32_t size = 0;
+    };
+
+    template<typename T>
```

Comment:
This tiny function object could be generalized and moved into utils.

Reply:
good idea -- will do in another pr.